Sources/ProjectDescription/TestAction.swift

```diff
@@ -65,144 +65,49 @@ public struct TestAction: Equatable, Codable {
+                               preActions: [ExecutionAction] = [],
+                               postActions: [ExecutionAction] = []) -> Self
     {
-        self.init(
-            testPlans: nil,
-            targets: targets,
-            arguments: arguments,
-            configurationName: configurationName,
-            coverage: coverage,
-            codeCoverageTargets: codeCoverageTargets,
-            expandVariableFromTarget: expandVariableFromTarget,
-            preActions: preActions,
-            postActions: postActions,
-            diagnosticsOptions: diagnosticsOptions,
-            language: language.flatMap(SchemeLanguage.init(stringLiteral:)),
-            region: region
-        )
-    }
-
-    /// Initializes a new instance of a test action
-    /// - Parameters:
-    ///   - targets: List of targets to be tested.
-    ///   - arguments: Arguments passed to the process running the tests.
-    ///   - config: Configuration that should be used for building the test targets.
-    ///   - coverage: True to collect the test coverage results.
-    ///   - codeCoverageTargets: List of targets for which Xcode will collect the coverage results.
-    ///   - preActions: ist of actions to be executed before running the tests.
-    ///   - postActions: List of actions to be executed after running the tests.
-    ///   - diagnosticsOptions: Diagnostics options.
-    ///   - language: Language (e.g. "pl").
-    ///   - region: Region (e.g. "PL").
-    public init(targets: [TestableTarget],
-                arguments: Arguments? = nil,
-                config: PresetBuildConfiguration = .debug,
-                coverage: Bool = false,
-                codeCoverageTargets: [TargetReference] = [],
-                expandVariableFromTarget: TargetReference? = nil,
-                preActions: [ExecutionAction] = [],
-                postActions: [ExecutionAction] = [],
-                diagnosticsOptions: [SchemeDiagnosticsOption] = [.mainThreadChecker],
-                language: String? = nil,
-                region: String? = nil)
-    {
-        self.init(
-            testPlans: nil,
+        Self(
+            testPlans: [],
             targets: targets,
-            arguments: arguments,
-            configurationName: config.name,
-            coverage: coverage,
-            codeCoverageTargets: codeCoverageTargets,
-            expandVariableFromTarget: expandVariableFromTarget,
+            arguments: nil,
+            configurationName: configuration.name,
+            coverage: false,
+            codeCoverageTargets: [],
+            expandVariableFromTarget: nil,
             preActions: preActions,
             postActions: postActions,
-            diagnosticsOptions: diagnosticsOptions,
-            language: language.flatMap(SchemeLanguage.init(stringLiteral:)),
-            region: region
+            diagnosticsOptions: [.mainThreadChecker],
+            language: nil,
+            region: nil
```

Comment:
we'll need to offer an API that allows users to configure/specify some of those parameters that are applicable when using the targets style test action.

Reply:
My bad. I extended the static method to allow passing all the options.